framework/auth/core.py

```diff
@@ -164,10 +165,18 @@ def logged_in(self):
     def from_kwargs(cls, request_args, kwargs):
         user = request_args.get('user') or kwargs.get('user') or _get_current_user()
         private_key = request_args.get('view_only')
-
+        try:
+            # Avoid circular import
+            from website.project.model import PrivateLink
+            private_link = PrivateLink.find_one(
+                Q('key', 'eq', private_key)
```

Comment:
I believe we need to further constrain by `is_deleted` in order to preserve the behavior of the previous `has_anonymous_link` method.